src/dsp_tools/utils/shared.py

```diff
@@ -163,7 +163,7 @@ def try_network_action(
         try:
             if args and not kwargs:
                 return action(*args)
-            elif kwargs and not args:
+            elif not args and kwargs:
                 return action(**kwargs)
             elif args and kwargs:
                 return action(*args, **kwargs)
```

Comment:
I would propose re-writing this entire section to a `match` roundabout like this:

```python
try:
    match (args, kwargs):
        case ([], {}):
            return action()
        case  (_, {}):
            return action(*args)
        case ([], _):
            return action(**kwargs)
        case _:
            return action(*args, **kwargs)
```

and this could even be extracted to a function `_execute_action()`, so your function only does one thing, namely retry.

Reply:
Nora an me, we tried to do the same thing, but in a match case, `[]` catches empty list **and** non-empty ones. AFAIU, there is no way in a match statement to differentiate between an empty container and a non-empty one - except if we would say `case (args, kwargs) if args and not kwargs`, but then there's no point any more in translating the if-else into a match case!

Reply:
no, `case []` does not match a non-empty list, I'm sure of that

Reply:
![image](https://github.com/dasch-swiss/dsp-tools/assets/33053745/f2f9499c-cda8-4838-bdbb-ad93e97cde9e)